File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        int n=nums.size();
        int counter=0;
        for(int i=0;i<=n;i++){
            counter=0;
            for(int j=0;j<n;j++){
                if(nums[j]>=i){
                    counter++;
                }
            }
            if(counter==i){
                return counter;
            }
        }
        return -1;
    }
};

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cs
public class Solution {
    public int SpecialArray(int[] nums) {
        Array.Sort(nums);
        // iterate from arr length to 1
        int i=nums.Length;
        int numsGreaterOrEqual;
        while(i>0){
            numsGreaterOrEqual=0;
            foreach(var n in nums){
                if(n>=i)
                    numsGreaterOrEqual++;
            }
            if(numsGreaterOrEqual==i)
                return i;
            i--;
        }
        return -1;
    }
}